python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -703,7 +706,15 @@ class NearestNeighbors(Base,
         cdef BallCoverIndex[int64_t, float, uint32_t]* rbc_index = \
             <BallCoverIndex[int64_t, float, uint32_t]*> 0
 
-        if self.working_algorithm_ == 'brute':
+        fallback_to_brute = self.working_algorithm_ == "rbc" and \
+            n_neighbors > math.sqrt(self.X_m.shape[0])
+
+        if fallback_to_brute:
+            warnings.warn("sqrt(%s) < n_neighbors (%s). "
+                          "falling back to brute force search" %
+                          (self.X_m.shape[0], n_neighbors))
```

Comment:
I think this message could be a bit more verbose of why this causes the fallback to brute from rbc